### PR TITLE
@craigspaeth => Add ability to import modules via absolute paths

### PR DESCRIPTION
**Pull request by damassi**
Closes https://github.com/artsy/force/issues/942.

Since I'm about to do a long stint inside of Force I wanted to sweep back through and follow up on a few things I had started thinking about a while back, with absolute module resolution being no.1 (and server-side React being no.2  😄)

What this PR does is set up [babel-plugin-module-resolver](https://github.com/tleunen/babel-plugin-module-resolver) to take absolute paths encountered in code and remap them to their corresponding file-system location, with aliases set up in `.babelrc`. There is a similar setup in Prediction. When using ES6+, resolution works seamlessly across the server and client and tests, though CoffeeScript code, while importable, must internally resolve paths the old way. I'm sure this can be fixed (but right now is less of a priority).  

Mappings are inside of [`.babelrc`](https://github.com/artsy/force/compare/master...damassi:add-absolute-module-resolution?expand=1#diff-e56633f72ecc521128b3db6586074d2cR11) and point to root of project. Example:

`import foo from 'desktop/apps/foo'`
`import bar from 'mobile/apps/bar'`
`import metaphysics from 'lib/metaphysics.coffee'`

### CODE REVIEW

File: desktop/apps/auction/client/actions.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
-import analyticsHooks from '../../../lib/analytics_hooks.coffee'
-import metaphysics from '../../../../lib/metaphysics.coffee'
-import { filterQuery } from '../queries/filter'
-import { worksByFollowedArtists } from '../queries/works_by_followed_artists'
+import analyticsHooks from 'lib/analytics_hooks.coffee'
+import metaphysics from 'root/lib/metaphysics.coffee'
+import { filterQuery } from 'apps/auction/queries/filter'
+import { worksByFollowedArtists } from 'apps/auction/queries/works_by_followed_artists'
```
Comment (damassi) [inline desktop/apps/auction/client/actions.js]: Example of client-side code

File: desktop/apps/auction/routes.js [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
-import metaphysics from '../../../lib/metaphysics'
-import { getLiveAuctionUrl } from '../../../utils/domain/auctions/urls'
+import Articles from 'collections/articles.coffee'
+import ArticlesQuery from 'apps/auction/queries/articles'
+import Auction from 'models/auction.coffee'
+import MeQuery from 'apps/auction/queries/me'
+import SaleQuery from 'apps/auction/queries/sale'
+import footerItems from 'apps/auction/footer_items'
+import metaphysics from 'root/lib/metaphysics'
+import { getLiveAuctionUrl } from 'root/utils/domain/auctions/urls'
```
Comment (damassi) [inline desktop/apps/auction/routes.js]: Example of server-side code

**Comment by alloy** [comment]
Dope 👌 

While I agree that most/all future dev will occur in desktop, I think that aliasing that to be the root path may lead to some confusion when reading code, I’d prefer we be explicit about it and get rid of it once we get rid of the mobile dir.

**Comment by craigspaeth** [comment]
Nice! I agree with Eloy about ./desktop not being the root—but for a different reason in that we're extracting shared mobile/desktop libs into the [root lib folder](https://github.com/artsy/force/tree/master/lib) and it would be nice to be able to `import metaphysics from 'lib/metaphysics'`

**Comment by damassi** [comment]
Ok sounds good! Will keep everything explicit (desktop/apps/whatevs), etc

**Comment by damassi** [comment]
Updated the root path to `./`.

**Pull request merged**